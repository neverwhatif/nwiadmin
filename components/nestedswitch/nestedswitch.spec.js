import React from 'react';
import { shallow } from 'enzyme';

import NestedSwitch, { renderRoute } from './index';

jest.mock('nwiadmin/scenes/404');

const routes = [{
    path: 'lorem',
    component: jest.fn(),
}, {
    path: 'ipsum',
    component: jest.fn(),
}];

describe('NestedSwitch', () => {
    const component = shallow(<NestedSwitch routes={routes} />);

    it('should render', () => {
        expect(component).toMatchSnapshot();
    });
    it('should render routes', () => {
        expect(component.find('Route').length).toBe(3); // 2 above, plus the error404
    });
    it('should use render prop for route', () => {
        const routeComponent = component.find('Route').first();
        const deepRoute = routeComponent.prop('render')({ lorem: 1 });

        expect(deepRoute.props.lorem).toBe(1);
    });
    it('should have working render prop for route', () => {
        const data = { ipsum: 1 };
        const deepRoute = renderRoute('div', { lorem: 1 }, 'lorem', data);
        const route = shallow(deepRoute);

        expect(route.type()).toBe('div');
        expect(route.prop('lorem')).toBe(1);
        expect(route.prop('basePath')).toBe('lorem');
        expect(route.prop('data')).toEqual(data);
    });
});
