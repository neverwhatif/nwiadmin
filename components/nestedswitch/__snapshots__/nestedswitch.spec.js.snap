// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NestedSwitch should render 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <NestedSwitch
    basePath=""
    routes={
      Array [
        Object {
          "component": [MockFunction],
          "path": "lorem",
        },
        Object {
          "component": [MockFunction],
          "path": "ipsum",
        },
      ]
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": Array [
        Array [
          <Route
            exact={false}
            path="/lorem"
            render={[Function]}
          />,
          <Route
            exact={false}
            path="/ipsum"
            render={[Function]}
          />,
        ],
        <Route
          component={[Function]}
        />,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": "lorem",
        "nodeType": "class",
        "props": Object {
          "exact": false,
          "path": "/lorem",
          "render": [Function],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": "ipsum",
        "nodeType": "class",
        "props": Object {
          "exact": false,
          "path": "/ipsum",
          "render": [Function],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "component": [Function],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          Array [
            <Route
              exact={false}
              path="/lorem"
              render={[Function]}
            />,
            <Route
              exact={false}
              path="/ipsum"
              render={[Function]}
            />,
          ],
          <Route
            component={[Function]}
          />,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": "lorem",
          "nodeType": "class",
          "props": Object {
            "exact": false,
            "path": "/lorem",
            "render": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "ipsum",
          "nodeType": "class",
          "props": Object {
            "exact": false,
            "path": "/ipsum",
            "render": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "component": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
